retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
Why do you need buffering to count that number? Should work as well with pass through.

I'm too busy to do deeper investigation here myself.
Can you present results of your investigation, like a high-frequency ingesting server before and after?

@fabxc perhaps you have an opinion on this?